apiserver/facades/agent/uniter/newlxdprofile.go

```diff
@@ -169,12 +169,17 @@ func (u *LXDProfileAPIv2) WatchInstanceData(args params.Entities) (params.Notify
 			result.Results[i].Error = apiservererrors.ServerError(apiservererrors.ErrPerm)
 			continue
 		}
-		machine, err := u.getLXDProfileMachineV2(tag)
+		unit, err := u.backend.Unit(tag.Id())
+		if err != nil {
+			result.Results[i].Error = apiservererrors.ServerError(err)
+			continue
+		}
+		machineID, err := unit.AssignedMachineId()
```

Comment:
We should add a TODO, because you could just get this information directly from the domain.
We could have a function called `WatchLXDProfilesForUnit`, this will correctly get the assigned machine ID based on the unit ID.

Reply:
It might be worth adding a task on unit epic for that.

Reply:
Sure, added that TODO.